media/js/mozorg/home/home.js

```diff
@@ -2,147 +2,173 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
-(function(Waypoint) {
-    'use strict';
-
-    function getNextEl(el) {
-        el = el.nextSibling;
-        while (el) {
-            if (el.nodeType === 1) {
-                return el;
-            }
-            el = el.nextSibling;
-        }
-        return null;
-    }
-
-    /*
-     * Video Tracking
-     */
-
-    function trackVideoInteraction(title, state) {
-        window.dataLayer.push({
-            'event': 'video-interaction',
-            'videoTitle': title,
-            'interaction': state
-        });
-    }
-
-    function initVideoEvents() {
-        var videos = document.querySelectorAll('video');
-        var videoCards = document.querySelectorAll('.mzp-c-card.has-video-embed .mzp-c-card-block-link');
+ /* global YT */
+ /* eslint no-unused-vars: [2, { "varsIgnorePattern": "onYouTubeIframeAPIReady" }] */
+
+// YouTube API hook has to be in global scope
+ function onYouTubeIframeAPIReady() {
+     'use strict';
+ }
+
+ (function(Waypoint) {
+     'use strict';
+
+     var tag = document.createElement('script');
+     tag.src = 'https://www.youtube.com/iframe_api';
+     var firstScriptTag = document.getElementsByTagName('script')[0];
+     firstScriptTag.parentNode.insertBefore(tag, firstScriptTag);
+
+     function getNextEl(el) {
+         el = el.nextSibling;
+         while (el) {
+             if (el.nodeType === 1) {
+                 return el;
+             }
+             el = el.nextSibling;
+         }
+         return null;
+     }
+
+     function initVideoEvents() {
+         var videoCards = document.querySelectorAll('.mzp-c-card.has-video-embed .mzp-c-card-block-link');
+
+         function playVideo(event) {
+             var card = event.currentTarget;
+             var title = card.querySelector('.mzp-c-card-title').innerText;
+             var sibling = getNextEl(event.currentTarget);
+             var content = sibling.querySelector('.mzp-c-card-video-content');
+
+             var videoLink = document.querySelector('.video-play');
+             var videoId = videoLink.getAttribute('data-id');
+             var player;
+
+             player = new YT.Player(videoLink, {
+                 height: '703',
+                 width: '1200',
+                 videoId: videoId,
+                 playerVars: {
+                     modestbranding: 1, // hide YouTube logo.
+                     rel: 0, // do not show related videos on end.
+                 },
+                 events: {
+                     'videoInteraction': trackVideoInteraction,
```

Comment:
`videoInteraction` isn't a supported event for youtube's API. You'll need to handle tracking GA events using the `onStateChange` event handler.

To get the video to auto-play, it looks like you'll also need to add an `onReady` event handler here also:

```javascript
'onReady': onPlayerReady,
```

And then something like:

```javascript
function onPlayerReady(event) {
    event.target.playVideo();
}
```

Reply:
The issue I was running into with `onPlayerReady()` was it was being called even _after_ the modal was closed. So I believe the start/stop needs to live in `onCreate:` and `onDestroy:` so it's dependent on the video modal. Wdyt?

Reply:
Hmm, right. I figured we could only call the YouTube API after the modal was created so we’d still need to wait for the player to be in the ready state before trying to play? As for stopping it when the modal closes, would calling `destroy()` on the video work?

Reply:
I found an older example of using this API inside a modal which might also help 😀

https://github.com/mozilla/bedrock/blob/master/media/js/firefox/dev-firstrun.js